### DISCUSSION TITLE
Implement minimal AST in SWC to skip acorn parsing in webpack

### DISCUSSION THREAD

**Discussion by timneutkens** [vercel/next.js#62372] [Help]
- [x] Implement `.webpackAST` method for next-swc
- [x] Pass js file contents to webpack so that it no longer needs to read them using `fs.readFile`
- [x] Implement loader that runs against `node_modules` that provides the AST to webpack
- [ ] Add webpack AST to next-swc transform() result
	- [ ] Implement passing the AST to webpack in next-swc-loader
- [ ] Ensure `webpackAST` passes all tests

**Comment by samcx** [comment]
Moving this to <picture data-single-emoji=":nextjs:" title=":nextjs:"><img class="emoji" width="20" height="auto" src="https://emoji.slack-edge.com/T0CAQ00TU/nextjs/2a85d633ae594556.png" alt=":nextjs:" align="absmiddle"></picture> [discussions](https://github.com/vercel/next.js/discussions) since this is technically not an issue!
